src/setup-open.ts

```diff
@@ -119,12 +119,13 @@ export function setupOpen(QuickSQLite: ISQLite) {
                 await hooks?.lockAcquired?.();
                 const res = await callback(context);
 
-                // Ensure that we only resolve after locks are freed
-                _.defer(() => resolve(res));
+                closeContextLock(dbName, id);
+                resolve(res)
               } catch (ex) {
-                _.defer(() => reject(ex));
+                closeContextLock(dbName, id);
```

Comment:
You should probably remove the call to `closeContextLock` on line 73 to avoid closing the lock twice.

Reply:
Then it won't close the lock of there was an error in the callback? It will skip past that straight to the catch block?

Reply:
Errors are already caught inside the callback. For the case of an exception (which shouldn't really propagate to that level), the `clsoeContextLock` call could be moved out of the finally block and in to the `catch` block on line 70